### PR TITLE
Improve sorting UX in M2A

### LINKED ISSUE #21532: When there is paging on a M2A it will remove sorting.
### Describe the Bug

When you use the Builder (M2A). and you have more than (the specified items per page) it will add a new page of items. 
Unfortunately it loses sorting and set the content back at when it's added. 

When adding items and you can sort as much as you want. Upon saving and the pagination kicks in it will remove the sorting.

### To Reproduce

To reproduce this you add a builder (M2A) to a collection and go to the page and add as much items as possible and set a sorting.
When done check if your sorting still works. 
If you have less than let's say default 15 change it to 10 and you'll see your sorting is gone.

### Video

https://github.com/directus/directus/assets/14160793/9945e58a-0510-43ad-8721-b744392b22f1

### Directus Version

v10.9.3

### Hosting Strategy

Self-Hosted (Docker Image)

### PR DESCRIPTION
**Pull request by hanneskuettner**
## Scope

What's changed:

- Added a notice that is shown if sorting in a M2A Builder is disabled due to having too many items for one page
- Added page size selector that is always shown if there are more than one page for the default page size, with the sizes `10, 25, 50, 100` and the configured page size as possible options
  - Page sizes smaller than the configured page size are not shown
  - Page sizes larger than the query limit are not shown
  - When selecting different sizes the 

Notice and disabled sorting
Screenshot 2024-04-17 at 14 26 04

Enabled sorting if page size > # of items
Screenshot 2024-04-17 at 14 26 19

## Potential Risks / Drawbacks

None

## Review Notes / Questions

- I would like feedback on the placement of the page size selector and its behavior as well as the wording of the notice.

---

Fixes #21532

### CODE REVIEW

File: app/src/composables/use-page-size.ts [hunk 1]
```diff
@@ -12,10 +12,10 @@ export function usePageSize<T = any>(
 
 	const pageSizes = computed<T[]>(() => {
 		if (queryLimit === undefined || queryLimit.max === -1) {
-			return availableSizes.map(mapCallback);
+			return toValue(availableSizes).map(mapCallback);
```
[the remark this answers is not in the document]
Comment (hanneskuettner) [reply app/src/composables/use-page-size.ts:15]: Sure thing!

**Review by paescuj — COMMENTED** [review]
Great work! 🏅 

Just a small note: Wondering if we should add another check to the pagination component to only display it if there are more than one page?

**Comment by hanneskuettner** [comment]
I thought about that as well, but then the page size select was jumping around even. Let me give it a try again and see how I feel about it then.

**Comment by hanneskuettner** [comment]
One more UX consideration. Should we persist the selected page size per collection? Right now it resets with every item you open. I'm not super sure about that one, since it kind of circumvents the Page Size configuration we have in the M2A options, and this is more of a stop-gap measure. On the other hand I like if stuff is persisted and consistent between pages.

**Comment by paescuj** [comment]
> I thought about that as well, but then the page size select was jumping around even more. Let me give it a try again and see how I feel about it then.

Yeah, let's keep it like this for now 👍 

> One more UX consideration. Should we persist the selected page size per collection:field? Right now it resets with every item you open. I'm not super sure about that one, since it kind of circumvents the Page Size configuration we have in the M2A options, and this is more of a stop-gap measure. On the other hand I like if stuff is persisted and consistent between pages.

We might indeed consider storing this in local storage, but if so let's address in a separate PR 🚀

**Review by paescuj — APPROVED** [review]

**Pull request merged**